src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CachedCaseUserRepository.java

```diff
@@ -43,4 +44,11 @@ public void revokeAccess(Long caseId, String userId, String caseRole) {
     public List<String> findCaseRoles(final Long caseId, final String userId) {
         return caseUserRoles.computeIfAbsent(caseId + userId, e -> caseUserRepository.findCaseRoles(caseId, userId));
     }
+
+    @Override
+    public List<CaseUserEntity> findCaseUserRoles(List<Long> caseIds, List<String> userIds) {
+        return caseAssignedUserRoles.computeIfAbsent(
+            new CaseAssignedUserRolesKey(caseIds, userIds),
+            e -> caseUserRepository.findCaseUserRoles(caseIds, userIds));
+    }
```

Comment:
This caching doesn't seem to be ok. What if the case-user-role assignments change? And what if the list of user ids and case ids are provided in a different order? And, what is the eviction policy of this cache?